homeassistant/components/sonos/alarms.py

```diff
@@ -0,0 +1,70 @@
+"""Class representing Sonos alarms."""
+from __future__ import annotations
+
+from collections.abc import Iterator
+import logging
+
+from pysonos import SoCo
+from pysonos.alarms import Alarm, get_alarms
+from pysonos.exceptions import SoCoException
+
+from homeassistant.core import HomeAssistant
+from homeassistant.helpers.dispatcher import async_dispatcher_send
+
+from .const import DATA_SONOS, SONOS_ALARMS_UPDATED, SONOS_CREATE_ALARM
+from .household_coordinator import SonosHouseholdCoordinator
+
+_LOGGER = logging.getLogger(__name__)
+
+
+class SonosAlarms(SonosHouseholdCoordinator):
+    """Storage class for Sonos alarms."""
+
+    def __init__(self, hass: HomeAssistant, household_id: str) -> None:
+        """Initialize the data."""
+        super().__init__(hass, household_id)
+        self._alarms: dict[str, Alarm] = {}
+
+    def __iter__(self) -> Iterator:
+        """Return an iterator for the known alarms."""
+        alarms = list(self._alarms.values())
+        return iter(alarms)
+
+    def get(self, alarm_id: str) -> Alarm | None:
+        """Get an Alarm instance."""
+        return self._alarms.get(alarm_id)
+
+    async def async_update_entities(self, soco: SoCo) -> bool:
+        """Create and update alarms entities, return success."""
+        new_alarms = await self.hass.async_add_executor_job(self.update_cache, soco)
+        if new_alarms is None:
+            return False
+
+        for alarm in new_alarms:
+            speaker = self.hass.data[DATA_SONOS].discovered[alarm.zone.uid]
+            async_dispatcher_send(self.hass, SONOS_CREATE_ALARM, speaker, [alarm])
```

Comment:
Do we really need to pass the alarm itself? Why not just pass the alarm_id and fetch the alarm from `hass.data`

Reply:
It can. I think during development I had been using multiple attributes in the entity constructor, but that's no longer the case. Will adjust. 👍